tests/integration/precompiles/staking/test_integration.go

```diff
@@ -1896,6 +1896,58 @@ func TestPrecompileIntegrationTestSuite(t *testing.T, create network.CreateEvmAp
 						txSenderFinalBal := balRes.Balance
 						Expect(txSenderFinalBal.Amount).To(Equal(txSenderInitialBal.Amount.Sub(fees)), "expected tx sender balance to be deducted by fees")
 					})
+					It("should test nested precompile calls with revert", func() {
+						outerTimes := int64(3)
+						innerTimes := int64(2)
+						expectedDelegations := outerTimes + 2 // 1 before + outerTimes after catches + 1 after loop
+						expectedDelegationAmount := math.NewInt(10).MulRaw(expectedDelegations)
+
+						callArgs := testutiltypes.CallArgs{
+							ContractABI: stakingReverterContract.ABI,
+							MethodName:  "nestedTryCatchDelegations",
+							Args: []interface{}{
+								big.NewInt(outerTimes), big.NewInt(innerTimes), s.network.GetValidators()[0].OperatorAddress,
+							},
+						}
+
+						expEvents := make([]string, 0, expectedDelegations)
+						for i := int64(0); i < expectedDelegations; i++ {
+							expEvents = append(expEvents, staking.EventTypeDelegate)
+						}
```

Comment:
2 ways to reduce code here, for funsies:
```suggestion
						expEvents := make([]string, 0, expectedDelegations)
						for range expectedDelegations {
							expEvents = append(expEvents, staking.EventTypeDelegate)
						}
```
or you can do

```suggestion
						slices.Repeat([]string{staking.EventTypeDelegate}, int(expectedDelegations))
```